docs/sphinx/source/installation.rst

```diff
@@ -34,49 +33,153 @@ There are many ways to install Python on your system, but the Anaconda
+`Fork A Repo <https://help.github.com/articles/fork-a-repo/>`_ for
+more details.
 
+.. _setupenvironment:
 
-Working at the bleeding edge
-----------------------------
+Set up a virtual environment
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
 We strongly recommend working in a `virtual environment
 <http://astropy.readthedocs.org/en/latest/development/workflow/
-virtual_pythons.html>`_ if you're going to use the development versions
-of the code. There are many ways to use virtual environments in Python,
-but Anaconda again provides the easiest solution:
+virtual_pythons.html>`_ if you're going to use an editable version
+of the library. You can skip this step if:
 
-#. Create a new conda environment for pvlib and pre-install a
-   handful of packages into the environment:
-   ``conda create --name pvlibdev python pandas scipy``
-#. Activate the new environment: ``source activate pvlibdev``
-#. Install the latest development version of pvlib:
+#. You already have Anaconda or another scientific Python distribution
+#. You don't mind polluting your Python installation with your
+   development version of pvlib.
+#. You don't want to work with multiple versions of pvlib.
 
-  #. If you don't plan to modify the source-code:
-     ``pip install git+https://github.com/pvlib/pvlib-python.git``
-  #. If you do plan to modify the source code:
-     Use the GitHub GUI application or git command-line tool to
-     clone this repository to your computer, then navigate your
-     command-line to the top-level pvlib-python directory,
-     then ``pip install -e .``
+There are many ways to use virtual environments in Python,
+but Anaconda again provides the easiest solution. These are often
+referred to as *conda environments*, but they're the same for our purposes.
 
-#. You may also consider installing additional packages into your
-   development environment:
-   ``conda install jupyter ipython seaborn nose flake8``
+#. **Create** a new conda environment for pvlib and pre-install
+   the required packages into the environment:
+   ``conda create --name pvlibdev python pandas scipy``
+#. **Activate** the new conda environment: ``source activate pvlibdev``
+#. **Install** additional packages into your development environment:
+   ``conda install jupyter ipython matplotlib seaborn nose flake8``
 
 The `conda documentation
 <http://conda.pydata.org/docs/using/index.html>`_ has more information
-on how to use virtual environments. You can also add ``-h`` to most
-conda commands to get help (e.g. ``conda -h`` or ``conda env -h``)
+on how to use conda virtual environments. You can also add ``-h`` to most
+pip and conda commands to get help (e.g. ``conda -h`` or ``conda env -h``)
+
+.. _installsource:
+
+Install the source code
+~~~~~~~~~~~~~~~~~~~~~~~
+
+Good news -- installing the source code is the easiest part!
+
+#. In your terminal or shell, **navigate** to the top-level
+   pvlib-python directory.
+#. **Install** pvlib-python in "development mode" by running
```

Comment:
I am hoping to review this more in depth, (I read the document somewhat
quickly, and agree with DaCoEx that some of the explanations are even
better than the standard docs), but I wanted to mention that people might
need some help wrapping their head around git, generally, and this document
seems helpful to me with the amount of version control understanding I
started with (advanced beginner):
http://betterexplained.com/articles/aha-moments-when-learning-git/



On Mon, Mar 21, 2016 at 5:55 AM, DaCoEx <notifications@github.com> wrote:

> In docs/sphinx/source/installation.rst
> <https://github.com/pvlib/pvlib-python/pull/138#discussion_r56816913>:
>
> >
> >  The `conda documentation
> >  <http://conda.pydata.org/docs/using/index.html>`_ has more information
> > -on how to use virtual environments. You can also add ``-h`` to most
> > -conda commands to get help (e.g. ``conda -h`` or ``conda env -h``)
> > +on how to use conda virtual environments. You can also add ``-h`` to most
> > +pip and conda commands to get help (e.g. ``conda -h`` or ``conda env -h``)
> > +
> > +.. _installsource:
> > +
> > +Install the source code
> > +~~~~~~~~~~~~~~~~~~~~~~~
> > +
> > +Good news -- installing the source code is the easiest part!
> > +
> > +#. In your terminal or shell, **navigate** to the top-level
> > +   pvlib-python directory.
> > +#. **Install** pvlib-python in "development mode" by running
>
> What I sometimes do not understand with development mode and the git
> branching concept where all branches are the same directory but change with
> checkout command:
>
> How do I keep the symlinks the editable working even if I work on
> different branches:
>
> Say I work on 1 feature so git checkout -b pvlib_newfeature. How do I
> prevent this editable environment from getting broken?
> Or should I, in this case, create a new environment for each branch.with a
> separate clone in another directory?
>
> Note: this is not a flaw of the text. so a link to a nice page that
> explains this would be useful.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/pvlib/pvlib-python/pull/138/files/2d6c1a83ab7debbb9b71e8aab9c9929898c2739e#r56816913>
>